src/main/java/service/operation/BalanceOperation.java

```diff
@@ -0,0 +1,19 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        for (Map.Entry<String, Integer> fruit : Storage.fruitsStorage.entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                fruit.setValue(fruitTransaction.getQuantity());
+                return;
+            }
+        }
+        Storage.fruitsStorage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+    }
```

Comment:
Consider encapsulating the logic of updating or adding a new balance into a private method to enhance readability and maintainability of the code.